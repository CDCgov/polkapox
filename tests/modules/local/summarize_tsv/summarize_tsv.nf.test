nextflow_process {

    name "Test Process SUMMARIZE_TSV"
    script "modules/local/summarize_tsv/summarize_tsv.nf"
    process "SUMMARIZE_TSV"

    test("Should run without failures") {

        when {
            params {
                af_cutoff     = 0.01
                DP_cutoff     = 5
                altDP_cutoff  = 3
                altQ_cutoff   = 20
                outdir        = "."
            }
            process {
                """
                input[0] = Channel.of(
                    [
                        meta: [id: 'test_sub'], ivar_tsv: file("${projectDir}/tests/nf-test-data/test_sub.ivar.tsv", checkIfExists: true)
                    ]
                )
                input[1] = Channel.of(coord1_start: '39086', coord1_end: '40204', coord2_start: '51427', coord2_end: '54447')
                input[2] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
