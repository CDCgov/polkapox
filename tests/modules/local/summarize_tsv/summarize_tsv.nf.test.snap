{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sub"
                        },
                        "test_sub_ivar_summary.txt:md5,a852dca08d151c5e78014321d0dada3a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5f7ff2b6c7b5fe12e5c5bf1cd132705a"
                ],
                "vars": [
                    [
                        {
                            "id": "test_sub"
                        },
                        "test_sub_ivar_summary.txt:md5,a852dca08d151c5e78014321d0dada3a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5f7ff2b6c7b5fe12e5c5bf1cd132705a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:12:18.137315115"
    }
}